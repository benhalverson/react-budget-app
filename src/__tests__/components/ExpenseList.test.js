import moment from 'moment';
import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import ConnectedExpenseList, { ExpenseList } from '../../components/ExpenseList';

const expenses = [
  {
    id: '1',
    description: 'Gum',
    note: '',
    amount: 195,
    createdAt: 0
  },
  {
    id: '2',
    description: 'Rent',
    note: '',
    amount: 109500,
    createdAt: moment(0)
      .subtract(4, 'days')
      .valueOf()
  },
  {
    id: '3',
    description: 'Credit Card',
    note: '',
    amount: 4500,
    createdAt: moment(0)
      .add(4, 'days')
      .valueOf()
  }
];


test('should render ExpenseList with expenses', () => {
  const wrapper = shallow(<ExpenseList expenses={expenses} />);

  expect(wrapper).toMatchSnapshot();
})

test('should render ExpenseList with expenses from redux', () => {
  const mockStore = configureStore();
  const store = mockStore({ expenses, filters: { text: '' } });
  const wrapper = shallow(<ConnectedExpenseList store={store} />);

  expect(wrapper.props().expenses).toEqual(expenses);
});

test('should render ExpenseList with empty message', () => {
  const wrapper = shallow(<ExpenseList expenses={[]}/>);
  expect(wrapper).toMatchSnapshot();
})