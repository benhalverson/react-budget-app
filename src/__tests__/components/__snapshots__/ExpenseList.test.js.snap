// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList with empty message 1`] = `ShallowWrapper {}`;

exports[`should render ExpenseList with expenses 1`] = `ShallowWrapper {}`;
